code/datums/components/mood.dm

```diff
@@ -356,5 +357,10 @@
 	else
 		clear_event(null, "area")
 
+/datum/component/mood/proc/on_revive(datum/source, full_heal)
+	if(!full_heal)
+		return
+	setSanity(initial(sanity))
```

Comment:
move remove temp moodies proc here instead of where it was before

Reply:
done